### PR TITLE
Move condition to bonus calculation.

### PR DESCRIPTION
**Pull request by vondele**
about 0.5% speedup.

No functional change.

### CODE REVIEW

**Comment by gogamoga1** [comment]
PGO
Results for 20 tests for each version:

            Base      Test      Diff      
    Mean    1698753   1703263   -4510     
    StDev   5365      10679     12406     

p-value: 0,642
speedup: 0,003

File: src/search.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ namespace {
   // History and stats update bonus, based on depth
   Value stat_bonus(Depth depth) {
     int d = depth / ONE_PLY ;
-    return Value(d * d + 2 * d - 2);
+    return d > 17 ? Value(0) : Value(d * d + 2 * d - 2);
```
Comment (mcostalba) [inline src/search.cpp]: Please use VALUE_ZERO here instead of Value(0)

**Comment by mcostalba** [comment]
@vondele this is more worth as a simplification than as a speed-up: 0.3% speed-up it is well below noise measurement level so gives no information. Instead as a simplification this patch makes sense IMO.

**Comment by vondele** [comment]
@mcostalba done. Sure, it is also a simplification.

**Pull request closed without merging**